### ISSUE TITLE
Client Crash When Whois From Another Client Connected To ZNC

### ISSUE THREAD

**Issue by vectr0n** [thelounge/lounge#933]
Ran into this crash when I ran a ``/whois <nick>`` in Hexchat that is also connected to the same ZNC instance as TheLounge. It spat out this when it crashed:

```
Feb 18 15:02:25 <removed> lounge[75171]: /usr/lib/node_modules/thelounge/node_modules/irc-framework/src/commands/handlers/user.js:89
Feb 18 15:02:25 <removed> lounge[75171]:             cache.nick = nick;
Feb 18 15:02:25 <removed> lounge[75171]:                          ^
Feb 18 15:02:25 <removed> lounge[75171]: ReferenceError: nick is not defined
Feb 18 15:02:25 <removed> lounge[75171]:     at IrcCommandHandler.RPL_ENDOFWHOIS (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/commands/handlers/user.js:89:26)
Feb 18 15:02:25 <removed> lounge[75171]:     at IrcCommandHandler.executeCommand (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/commands/handler.js:62:37)
Feb 18 15:02:25 <removed> lounge[75171]:     at IrcCommandHandler.dispatch (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/commands/handler.js:49:14)
Feb 18 15:02:25 <removed> lounge[75171]:     at /usr/lib/node_modules/thelounge/node_modules/irc-framework/src/client.js:103:36
Feb 18 15:02:25 <removed> lounge[75171]:     at next (/usr/lib/node_modules/thelounge/node_modules/middleware-handler/index.js:42:18)
Feb 18 15:02:25 <removed> lounge[75171]:     at MiddlewareHandler.handle (/usr/lib/node_modules/thelounge/node_modules/middleware-handler/index.js:64:3)
Feb 18 15:02:25 <removed> lounge[75171]:     at Connection.<anonymous> (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/client.js:97:31)
Feb 18 15:02:25 <removed> lounge[75171]:     at Connection.emit (/usr/lib/node_modules/thelounge/node_modules/eventemitter3/index.js:130:35)
Feb 18 15:02:25 <removed> lounge[75171]:     at Connection.processReadBuffer (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/connection.js:265:14)
Feb 18 15:02:25 <removed> lounge[75171]:     at Connection.socketLine (/usr/lib/node_modules/thelounge/node_modules/irc-framework/src/connection.js:99:14)
Feb 18 15:02:25 <removed> systemd[1]: thelounge.service: Main process exited, code=exited, status=1/FAILURE
Feb 18 15:02:25 <removed> systemd[1]: thelounge.service: Unit entered failed state.
Feb 18 15:02:25 <removed> systemd[1]: thelounge.service: Failed with result 'exit-code'.
```

I can reproduce the crash every single time.

**Comment by xPaw** [comment]
Unable to reproduce. What lounge version are you running? Did you run `npm install`?

**Comment by vectr0n** [comment]
Updated with new info, installed from npm.

**Comment by dgw** [comment]
I cannot reproduce on v2.2.1 installed from npm.

**Comment by xPaw** [comment]
However I do see why that issue *might* happen, it's trying to assign something that doesn't exist. https://github.com/kiwiirc/irc-framework/blob/fc6f4c7a19b305f872e5aeabac3f6f148e5f1556/src/commands/handlers/user.js#L89

That's a bug in irc-fw, cc @prawnsalad.

**Comment by vectr0n** [comment]
If you whois a nickname that is offline from Hexchat it crashes the instance with the above error. If the nickname is online there is no issue.

**Comment by xPaw** [comment]
Fixed in https://github.com/kiwiirc/irc-framework/pull/82

**Comment by vectr0n** [comment]
[03:02:25 PM] * irc.<removed>.com vectr0n_test :No such nick/channel
[03:02:25 PM] * [vectr0n_test] End of WHOIS list.

As you can see the time stamps match from the crash and when I did the whois.

**Comment by vectr0n** [comment]
Thanks @xPaw

**Comment by astorije** [comment]
FYI, this issue has been fixed in `irc-framework` so it will make it to The Lounge whenever @prawnsalad cuts a release 🎉. Labelling as v2.2.2 as I'm pretty sure `irc-framework` release will happen sooner than ours.

**Comment by xPaw** [comment]
#918 should have fixed this.

**Comment by astorije** [comment]
@xPaw, I believe `irc-framework` 2.6.0 did not fix this actually, as you can see your fix in the 2.6.1 diff: https://github.com/kiwiirc/irc-framework/compare/v2.6.0...v2.6.1.

Problem is, at this time 2.6.0 is still [the latest version of the npm package](https://www.npmjs.com/package/irc-framework) so that's why Greenkeeper did not warn us of an update. I'm wondering if @prawnsalad didn't forget to publish to npm after pushing the tag.

**Comment by astorije** [comment]
Re-opening until we bump `irc-framework` to version 2.6.1.
